qiskit/aqua/algorithms/phase_estimators/hamiltonian_pe.py

```diff
@@ -0,0 +1,129 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Phase estimation for the spectrum of a Hamiltonian"""
+
+from typing import Optional, Union
+from qiskit import QuantumCircuit
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.operators import EvolutionBase, OperatorBase
+from qiskit.providers import BaseBackend, Backend
+from .phase_estimation import PhaseEstimation
+from . import phase_estimation_scale
+from .hamiltonian_pe_result import HamiltonianPEResult
+from .phase_estimation_scale import PhaseEstimationScale
+
+
+class HamiltonianPE(PhaseEstimation):
```

Comment:
Should we call this `HamiltonianPhaseEstimation` in full. I am wondering about having this as PE, when we spell out PhaseEstimation for the other algorithm in full elsewhere.